src/lib.rs

```diff
@@ -0,0 +1,230 @@
+    ) -> UwUify<'a> {
+        // I dislike this
+
+        let mut linkify = LinkFinder::new();
+        linkify.kinds(&[LinkKind::Email, LinkKind::Url]);
+        linkify.url_must_have_scheme(false);
+        let mut uwuify = UwUify {
+            text: text.unwrap_or_default(),
+            input: infile.unwrap_or_default(),
+            output: outfile.unwrap_or_default(),
+            linkify,
+            ..Default::default()
+        };
+
+        if random {
+            uwuify.floating_rng = Xoshiro256Plus::seed_from_u64(rand::rngs::OsRng.next_u64());
+            uwuify.int_rng = Xoshiro256PlusPlus::seed_from_u64(rand::rngs::OsRng.next_u64());
+        }
+
+        if let Some(words) = words {
+            uwuify.words = words.parse::<f64>().unwrap();
+        }
+        if let Some(faces) = faces {
+            uwuify.faces = faces.parse::<f64>().unwrap();
+        }
+        if let Some(actions) = actions {
+            uwuify.actions = actions.parse::<f64>().unwrap();
+        }
+        if let Some(stutters) = stutters {
+            uwuify.stutters = stutters.parse::<f64>().unwrap();
+        }
+        uwuify
+    }
+
+    pub fn uwuify(&mut self) -> Result<(), Error> {
+        // Handle Text
+        if !self.text.is_empty() {
+            // Handle Text Output
+            if !self.output.is_empty() {
+                if Path::new(&self.output).exists() {
+                    return Err(Error::new(
+                        std::io::ErrorKind::AlreadyExists,
+                        format!("File '{}' already exists, aborting operation", &self.output),
+                    ));
+                }
+
+                let mut uwu_out_file = BufWriter::new(File::create(&self.output)?);
+                let uwu_progress_bar = progress_bar!();
+                self.uwuify_sentence(self.text, &mut uwu_out_file)?;
+
+                uwu_progress_bar.finish_with_message("UwU'ifying Complete!");
+                Ok(())
+            } else {
+                #[cfg(not(test))]
+                let stdout = std::io::stdout();
+                #[cfg(not(test))]
+                let mut out = BufWriter::new(stdout.lock());
+                #[cfg(test)]
+                let mut out = std::io::sink();
+                self.uwuify_sentence(self.text, &mut out)?;
+                #[cfg(not(test))]
+                out.write_all(b"\n")?;
+                Ok(())
+            }
+        } else {
+            // Handle File I/O
+            if Path::new(&self.output).exists() {
+                return Err(Error::new(
+                    std::io::ErrorKind::AlreadyExists,
+                    format!("File '{}' already exists, aborting operation", &self.output),
+                ));
+            }
+
+            let uwu_progress_bar = progress_bar!();
+            self.uwuify_sentence(
+                unsafe {
+                    std::str::from_utf8_unchecked(
+                        memmap::Mmap::map(&File::open(&self.input)?)?.as_ref(),
```

Comment:
For all uses of `?`, I believe the behaviour is that rust immediately panics? I avoided the use of `?` for this reason as it would hurt its robustness. Are we able to change it so that an error is bubbled back up to the main method?

Reply:
`?` is not the same as `unwrap()`, `?` returns the function with an `Err()` if the given result is also an `Err()`. `unwrap()` panics and stops everything immediately as soon as an error in encountered

Reply:
```rust
// This function returns Err("something went wrong")
fn error_fn_question() -> Result<(), &'static str> {
    let result: Result<(), &'static str> = Err("something went wrong");
    Ok(result?)
}

// This function panics and exits the program
fn error_fn_question() -> Result<(), &'static str> {
    let result: Result<(), &'static str> = Err("something went wrong");
    Ok(result.unwrap())
}

Reply:
Ahhh okay, I think I misunderstood this within the docs. I thought that it would always `panic` out. That's really helpful, thanks!